packages/node_modules/@webex/webex-core/test/unit/spec/services/interceptors/server-error.js

```diff
@@ -0,0 +1,161 @@
+        beforeEach(() => {
+          options.uri = 'http://not-a-url.com/';
+          reason = new WebexHttpError.InternalServerError({
+            message: 'test message',
+            statusCode: 500,
+            options: {
+              url: 'http://not-a-url.com/',
+              headers: {
+                trackingId: 'tid'
+              }
+            }
+          });
+        });
+
+        describe('when the web-ha feature is enabled', () => {
+          let get;
+          let markUrlFailedAndGetNew;
+          let submitClientMetrics;
+
+          beforeEach('mock device and metrics', () => {
+            interceptor.webex = {
+              internal: {
+                device: {
+                  features: {
+                    developer: {
+                      get: sinon.stub()
+                    }
+                  },
+                  markUrlFailedAndGetNew: sinon.stub()
+                },
+                metrics: {
+                  submitClientMetrics: sinon.spy()
+                }
+              }
+            };
+
+            get = interceptor.webex.internal.device.features.developer.get;
+            markUrlFailedAndGetNew = interceptor
+              .webex
+              .internal
+              .device
+              .markUrlFailedAndGetNew;
+            submitClientMetrics = interceptor
+              .webex
+              .internal
+              .metrics
+              .submitClientMetrics;
+
+            get.returns({value: true});
+            markUrlFailedAndGetNew.resolves();
+          });
+
+          it('should get the feature \'web-high-availability\'', () =>
+            interceptor.onResponseError(options, reason)
+              .then(() => assert.isTrue(false, 'returned a resolved promise'))
+              .catch(() => assert.calledWith(get, 'web-high-availability')));
+
+          it('should submit appropriate client metrics', () =>
+            interceptor.onResponseError(options, reason)
+              .then(() => assert.isTrue(false, 'returned a resolved promise'))
+              .catch(() =>
+                assert.calledWith(submitClientMetrics,
+                  'web-ha',
+                  {
+                    fields: {success: false},
+                    tags: {
+                      action: 'failed',
+                      error: reason.message,
+                      url: options.uri
+                    }
+                  })));
+
+          it('should mark a url as failed', () =>
+            interceptor.onResponseError(options, reason)
+              .then(() => assert.isTrue(false, 'returned a resolved promise'))
+              .catch(() =>
+                assert.calledWith(markUrlFailedAndGetNew, options.uri)));
+
+          it('should return a rejected promise with a reason', () => {
+
+          });
+        });
+
+        describe('when the web-ha feature is not available or disabled', () => {
```

Comment:
Does this need some sort of test setup? If it is the default config, you could remove this describe block and move the `it` up a level, around line 44.

Reply:
Organized the test setup to appear more reasonable, moved an `it()` to a better place.